src/5.1_CW/Shape.ts

```diff
@@ -0,0 +1,13 @@
+abstract class Shape {
+    public name: string;
+    public color: string;
+
+    constructor(color: string, name: string) {
+        this.color = color;
+        this.name = name;
+    }
+
+    public calculateArea(): number {
```

Comment:
Why go through the hassle of hard-coding method realizations just to override them in child components? Abstract methods offer a much cleaner and more efficient solution.